tests/storage/snapshot.go

```diff
@@ -75,6 +75,22 @@ var _ = SIGDescribe("[Serial]VirtualMachineSnapshot Tests", func() {
 		}, 180*time.Second, time.Second).Should(BeTrue())
 	}
 
+	waitSnapshotSucceeded := func(snapshotName string) (snapshot *snapshotv1.VirtualMachineSnapshot) {
+		Eventually(func() bool {
+			snapshot, err = virtClient.VirtualMachineSnapshot(vm.Namespace).Get(context.Background(), snapshotName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			return snapshot.Status != nil &&
+				len(snapshot.Status.Conditions) == 2 &&
+				snapshot.Status.Conditions[0].Status == corev1.ConditionFalse &&
```

Comment:
Are we guaranteed the order will remain the same all the time? Maybe better to loop over the conditions and check if we find the condition by type? This is very hard to read and understand right now.

Reply:
ill work on creating a more elegant function to be used from all tests, if it wont get in with this PR ill add it in a different one